### DISCUSSION TITLE
New Configuration Annotation Processor and support of Legacy way

### DISCUSSION THREAD

**Discussion by gcuisinier** [quarkusio/quarkus#42979] [Q&A] [answered]
Hi,

I am working on migrating our internal extension to Quarkus 3.14.
And the migration Guide for Extension using @ConfigRoot explain that it needed to configure the build to use the legacy behavior.

But does this legacy support is planned to be deprecated and purged in future ?
And so, does migrating to the new way ( ConfigRoot+ConfigMapping and Interface style ) is strongly recommended ?

[a comment by an automated account is not shown]

**Comment by radcortez** [comment, marked as answer]
> But does this legacy support is planned to be deprecated and purged in future ?

Yes, but it may still take a while.

> And so, does migrating to the new way ( ConfigRoot+ConfigMapping and Interface style ) is strongly recommended ?

Yes. There is no migration guide, but it should be straightforward. Check:
- https://quarkus.io/guides/writing-extensions#configuration
- https://quarkus.io/guides/config-mappings

**Reply by gsmet** [reply]
Yeah, I would add that there is really no hurry (thus why there is a simple workaround). We still have quite a lot of `@ConfigRoot` in the core and it will take time to migrate all of them - and only then we will start considering dropping the legacy `@ConfigRoot` class support.

So you can plan this work but you don't need to rush it.

**Reply by gsmet** [reply]
I added a note in the migration guide. Thanks for raising the question!

**Comment by gcuisinier** [comment]
Ok, thank you for the info !
